example/plugins/c-api/txn_data_sink/txn_data_sink.cc

```diff
@@ -117,43 +117,122 @@ client_reader(TSCont contp, TSEvent event, void *edata)
   return 0;
 }
 
+/** The handler for transaction data sink for response bodies. */
 int
-enable_agent_check(TSHttpTxn txnp)
+response_body_reader(TSCont contp, TSEvent event, void *edata)
 {
-  TSMBuffer req_buf;
-  TSMLoc req_loc;
-  int zret = 0;
+  SinkData *data = static_cast<SinkData *>(TSContDataGet(contp));
+
+  // If we got closed, we're done.
+  if (TSVConnClosedGet(contp)) {
+    delete data;
+    TSContDestroy(contp);
+    return 0;
+  }
+
+  TSVIO input_vio = TSVConnWriteVIOGet(contp);
 
+  if (data == nullptr) {
+    data = new SinkData;
+    TSContDataSet(contp, data);
+  }
+  return body_reader_helper(event, input_vio, data->response_body_bytes);
+}
+
+/** The handler for transaction data sink for request bodies. */
+int
+request_body_reader(TSCont contp, TSEvent event, void *edata)
```

Comment:
This is identical to the previous function except for the member in `data`. I'm not sure if that should be parameterized out or whether having the functions separate makes it a better example.

Reply:
Good point. For the example I think it's probably easier to understand the function as one piece instead of split across three. I consolidated the logic to the single function using a parameter as you suggested.